lib/ping.go

```diff
@@ -0,0 +1,67 @@
+package vole
+
+import (
+	"bytes"
+	"context"
+	"crypto/rand"
+	"fmt"
+	"time"
+
+	"github.com/libp2p/go-libp2p/core/network"
+	"github.com/libp2p/go-libp2p/core/peer"
+	"github.com/libp2p/go-libp2p/p2p/protocol/identify"
+)
+
+func Ping(ctx context.Context, direct bool, p *peer.AddrInfo) error {
+	if !direct {
+		// We don't want a direct connection, so set this to a high value so
+		// that we don't learn our public address
+		identify.ActivationThresh = 100
```

Comment:
Direct won't work until we merge https://github.com/ipfs-shipyard/vole/pull/39 since that sets this to 1 in init.